### PR TITLE
Adds armblade implant and support for item implants

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

title

## Changelog
:cl:
add: Unobtainable armblade implant
code: support for item-focused implants and bodypart-only implants
/:cl:

### CODE REVIEW

File: code/game/objects/items/implants/implant.dm [hunk 1]
```diff
@@ -54,6 +58,9 @@
 	SHOULD_CALL_PARENT(TRUE)
 	if(!ishuman(target))
 		return FALSE
+	if(!(user.zone_selected in allowed_limbs))
+		to_chat(user, "<span class='warning'>You cannot implant this into that limb!!</span>")
```
Comment (XSlayer300) [inline code/game/objects/items/implants/implant.dm:62]:
```suggestion
		to_chat(user, "<span class='warning'>You cannot implant this into that limb!</span>")
```

File: code/game/objects/items/implants/implant_items.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	icon_state = "armblade"
+	helditem = /obj/item/weapon/mantisblade
+
+/obj/item/implant/deployitem/blade/get_data()
+	return {"
+	<b>Implant Specifications:</b><BR>
+	<b>Name:</b> Nanotrasen MA-12 Mantis Implant<BR>
+	<HR>
+	<b>Implant Details:</b><BR>
+	<b>Function:</b> Upon activation deploys a large blade from the users arm.<BR>"}
```
Comment (XSlayer300) [inline code/game/objects/items/implants/implant_items.dm:55]:
```suggestion
	<b>Function:</b> Upon activation, the user deploys a large blade from the their arm.<BR>"}
```

File: code/game/objects/items/implants/implant_items.dm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	. = ..()
+	playsound(implant_owner, 'sound/weapons/wristblades_on.ogg', 15, 0, 1)
+
+/obj/item/implant/deployitem/blade/fetch_item()
+	. = ..()
+	playsound(implant_owner, 'sound/weapons/wristblades_off.ogg', 15, 0, 1)
+
+/obj/item/weapon/mantisblade
+	name = "mantis arm blade"
+	desc = "A wicked-looking folding blade capable of being concealed within a humans arm."
```
Comment (XSlayer300) [inline code/game/objects/items/implants/implant_items.dm:67]:
```suggestion
	desc = "A wicked-looking folding blade capable of being concealed within a human's arm."
```

File: code/game/objects/items/implants/implant_items.dm [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	put_in_slots()
+	RegisterSignal(helditem, COMSIG_ITEM_DROPPED, .proc/fetch_item)
+
+///Takes the held item and puts it into it's predestined slot
+/obj/item/implant/deployitem/proc/put_in_slots()
+	switch(part.name)
+		if(BODY_ZONE_R_ARM)
+			implant_owner.put_in_r_hand(helditem)
+		if(BODY_ZONE_L_ARM)
+			implant_owner.put_in_l_hand(helditem)
```
Comment (SplinterGP) [inline code/game/objects/items/implants/implant_items.dm:31]: should probably be able to put in other slots as well

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**